src/lifecycles/prop.helpers.js

```diff
@@ -26,10 +30,19 @@ export function getProps(appOrParcel) {
       customProps
     );
   }
+
   const result = assign({}, customProps, {
     name,
     mountParcel: mountParcel.bind(appOrParcel),
     singleSpa,
+    route: {
+      get params() {
```

Comment:
I would prefer not using a Javascript getter here, since it has performance implications just for accessing `props.route`. Ideally we'd calculate the route params once per URL change rather than every time any code does `props.route`. I haven't thought through whether it would be possible to do that, but it's something that should be explored. The `toDynamicPathRegexInfo` function is complex enough that I don't want it running frequently

Reply:
It may not be possible at the moment without using some kind of hack. The most naive approach to reduce the complexity may be memoization of `extractDynamicParams` but that may contribute to memory leak.

Reply:
I don't think there's a foul proof native way of listening to URL changes. We may try building `route` object whenever [`navigateToUrl`](https://github.com/single-spa/single-spa/blob/v5.9.5/src/navigation/navigation-events.js#L19) is invoked.

will it be good enough or we may try something else?

Reply:
@joeldenning

Reply:
@dsfx3d I don't think `navigateToUrl` is a good solution because it is not the only or even primary way people use to change the URL.  I think it should be done in an event handler that fires when the URL changes.  Just my $0.02.